waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -132,6 +160,22 @@ private void reconnectIfDisconnected() {
       }
     }
 
+    private void genToken() throws Throwable {
```

Comment:
I quite like the approach of the TokenWrappingHMSHandler being separate for this code the client I suggest we do the same and create a `SaslMetastoreClientHander implements  InvocationHandler`  
This class newInstance methos can check if useSasl=true and wrap the `ReconnectingMetastoreClientInvocationHandler` inside the Sasl handler. That way we don't affect the non sasl code paths at all.

Let me know what you think.